src/app/layout/add-contact/name-validator.ts

```diff
@@ -0,0 +1,10 @@
+import { AbstractControl, ValidationErrors } from '@angular/forms';  
+    
+export class NameValidator {  
+    static cannotContainSpace(control: AbstractControl) : ValidationErrors | null {  
+        if((control.value as string).indexOf(' ') >= 0){  
```

Comment:
Just be aware that `as string` doesn't cast the variable to a string. So if you were to put this validator onto a `<input type="number" />` then it would error